packages/arb-token-bridge-ui/src/util/isDepositMode.ts

```diff
@@ -8,15 +8,21 @@ export function isDepositMode({
   destinationChainId: number
 }) {
   const {
-    isEthereumMainnetOrTestnet: isSourceChainEthereum,
+    isEthereumMainnetOrTestnet: isSourceChainEthereumMainnetOrTestnet,
     isArbitrum: isSourceChainArbitrum,
     isBase: isSourceChainBase
   } = isNetwork(sourceChainId)
   const { isOrbitChain: isDestinationChainOrbit } =
     isNetwork(destinationChainId)
 
+  const validDestinationChains = getDestinationChainIds(sourceChainId)
+
+  if (!validDestinationChains.includes(destinationChainId)) {
+    throw new Error('Unsupported source and destination chain pair.')
+  }
+
   const isDepositMode =
-    isSourceChainEthereum ||
+    (isSourceChainEthereumMainnetOrTestnet && !isDestinationChainOrbit) ||
     isSourceChainBase ||
     (isSourceChainArbitrum && isDestinationChainOrbit)
```

Comment:
This is incorrect. We should also check for `partnerChainId`

L1 to orbit chain (settling to L1) should be a deposit, not teleport